src/dbnode/encoding/proto/custom_unmarshal.go

```diff
@@ -332,14 +344,15 @@ func unmarshalSimpleField(fd *desc.FieldDescriptor, v uint64) (unmarshalValue, e
 }
 
 func (u *customUnmarshaler) resetAndUnmarshal(schema *desc.MessageDescriptor, buf []byte) error {
-	if schema != u.schema {
-		u.nonCustomValues = dynamic.NewMessage(schema)
-		u.nonCustomValues.Reset()
-	}
-
 	u.schema = schema
 	u.numNonCustom = 0
 	u.customValues = u.customValues[:0]
```

Comment:
Does `customValues` also need its entries reset? (does it contain any pointer types?)